dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp

```diff
@@ -777,7 +780,18 @@ void DAGStorageInterpreter::buildLocalStreams(DAGPipeline & pipeline, size_t max
                 throw;
             }
         }
-        pipeline.streams.insert(pipeline.streams.end(), current_pipeline.streams.begin(), current_pipeline.streams.end());
+        if (has_multiple_partitions)
+            stream_pool->addPartitionStreams(current_pipeline.streams);
+        else
+            pipeline.streams.insert(pipeline.streams.end(), current_pipeline.streams.begin(), current_pipeline.streams.end());
+    }
+    if (has_multiple_partitions)
+    {
+        String req_info = dag_context.isMPPTask() ? dag_context.getMPPTaskId().toString() : "";
+        for (int i = 0; i < static_cast<int>(max_streams); ++i)
```

Comment:
The stream number should be `std::min(max_streams, the stream number in stream_pool)`?

Reply:
yes, nice find

Reply:
done